src/OnboardingSPA/components/Drawer/DrawerPanel/DesignThemeStylesPreview.js

```diff
@@ -14,36 +15,38 @@ const DesignThemeStylesPreview = () => {
 	const [ globalStyles, setGlobalStyles ] = useState();
 	const [ selectedStyle, setSelectedStyle ] = useState( '' );
 
-	const { currentStep, currentData, storedPreviewSettings } = useSelect(
-		( select ) => {
+	const { currentStep, currentData, storedPreviewSettings, themeStatus } =
+		useSelect( ( select ) => {
 			return {
 				currentStep: select( nfdOnboardingStore ).getCurrentStep(),
 				currentData:
 					select( nfdOnboardingStore ).getCurrentOnboardingData(),
 				storedPreviewSettings:
 					select( nfdOnboardingStore ).getPreviewSettings(),
+				themeStatus: select( nfdOnboardingStore ).getThemeStatus(),
 			};
-		},
-		[]
-	);
+		}, [] );
 
 	const { updatePreviewSettings, setCurrentOnboardingData } =
 		useDispatch( nfdOnboardingStore );
 
 	const getStylesAndPatterns = async () => {
-		const pattern = await getPatterns( currentStep.patternId, true );
-		const globalStyles = await getGlobalStyles();
-		setPattern( pattern?.body );
-		setGlobalStyles( globalStyles?.body );
-		let selectedStyle;
+		const patternResponse = await getPatterns(
+			currentStep.patternId,
+			true
+		);
+		const globalStylesResponse = await getGlobalStyles();
+		setPattern( patternResponse?.body );
+		setGlobalStyles( globalStylesResponse?.body );
```

Comment:
Why don't we have checks like:
`if ( patternsResponse?.error ) {
			return updateThemeStatus( THEME_STATUS_NOT_ACTIVE );
		}
		const globalStylesResponse = await getGlobalStyles();
		if ( globalStylesResponse?.error ) {
			return updateThemeStatus( THEME_STATUS_NOT_ACTIVE );
		}` 
		
over here too? Or can't the function getting the responses itself do checks like this so as to prevent duplicacy in every step??